src/main/webapp/script.js

```diff
@@ -0,0 +1,28 @@
+function getOriginalRecipe() {
+  const key = document.getElementById("key").value;
+  if (key) {
+    fetch("/new-recipe?key=" + key).then(response => response.json()).then((recipe) => {
+      populateRecipeCreationForm(recipe);
+    });
+  }
+}
+
+function populateRecipeCreationForm(recipe) {
+  var name = document.getElementById("name");
```

Comment:
Can you combine the two statements into one, since name is a single use variable. (Similarly for description?).

Please let me know if this doesn't work as intended. I tried it on Chrome console and it did, it might be different for loading from a script.